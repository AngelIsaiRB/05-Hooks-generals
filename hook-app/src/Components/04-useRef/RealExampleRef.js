import React, { useState } from 'react'
import "../02-useEfect/effects.css"
import { MultipleCustomHooks } from '../02-useEfect/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setshow] = useState(false)

    return (
        <div>
            <h1>real example ref</h1>
            <hr/>
            {show && <MultipleCustomHooks/>}
            <button
                className="btn btn-primary mt-5"
                onClick={()=>setshow(!show)}
            >
                Show/Hide
            </button>
        </div>
    )
}
