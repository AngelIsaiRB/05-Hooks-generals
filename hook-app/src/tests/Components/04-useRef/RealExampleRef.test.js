import { shallow } from 'enzyme';
import React from 'react'
import { RealExampleRef } from '../../../Components/04-useRef/RealExampleRef';



describe('pruebas en el componente <RealExampleRef>', () => {
    const wrapper = shallow(<RealExampleRef/>) ;

    test('should sow the component', () => {
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
    });

    test('should show the component <multipleCustomHooks/>', () => {
        wrapper.find("button").simulate("click");
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true); 
    })
    
    
})
