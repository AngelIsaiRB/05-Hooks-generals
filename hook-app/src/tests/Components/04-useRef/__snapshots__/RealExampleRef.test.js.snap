// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <RealExampleRef> should sow the component 1`] = `
<div>
  <h1>
    real example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
